first sign of DB

ilya-pauzner opened the pull request:
Первая пошла, граждане!

ilya-pauzner committed:
first sign of DB

it works btw

s17b2-voroneckij commented on start_me.py:
```diff
@@ -29,22 +29,30 @@ def _(place, *args, **kwargs):
 def api_init():
-    global api, access_token, names
+    global api, access_token, names, cursor
+    conn = sqlite3.connect('db.db')
```
Плохое название

ilya-pauzner replied:
Какое из плохое?

s17b2-voroneckij replied:
db.db

s17b2-voroneckij commented on start_me.py:
```diff
@@ -29,22 +29,30 @@ def _(place, *args, **kwargs):
     try:
         # DB!
-        access_token = open("token").read()
+        for token in cursor.execute('SELECT * from token_table'):
```
Проверить, что там один токен

ilya-pauzner replied:
Казалось бы, их там два появиться и не может.

s17b2-voroneckij commented on start_me.py:
```diff
@@ -29,22 +29,30 @@ def _(place, *args, **kwargs):
+    global api, access_token, names, cursor
+    conn = sqlite3.connect('db.db')
+    cursor = conn.cursor()
     try:
```
Вместо try надо проверять, что там есть токен

ilya-pauzner replied:
А как?
Вдруг там и таблицы такой нет?

s17b2-voroneckij replied:
гугли, читай доки.

s17b2-voroneckij commented on start_me.py:
```diff
@@ -29,22 +29,30 @@ def _(place, *args, **kwargs):
         vk_session = vk.AuthSession(app_id='4771271', user_login=login, user_password=password, scope=4096)
         access_token = vk_session.access_token
 
+        cursor.execute('CREATE TABLE token_table (value text)')
```
А если она и так есть?

ilya-pauzner replied:
Справедливо, но что с этим делать?
Создать при первом запуске пустую базу но со всеми таблицами и далее подразумевать что таблицы есть, возможно пустые?

s17b2-voroneckij replied:
Ну да, если таблицы нет - создать пустую и заполнрять

ilya-pauzner committed:
dict names now is id_to_name table in DB

ilya-pauzner committed:
almost done

ilya-pauzner committed:
db everywhere

some bugs to fix
--- some things are sorted as strings
--- try-except to get max from empty table
--- instead of looking through chats one after another, PROGRESSBAR, somehow get them all (study VK API)
--- chat names with spaces in them get {bereza vorona}
--- works really slow, why? maybe disable autocommit
--- instead of query1 + table_name + query2, something more clever
--- something else

ilya-pauzner committed:
db everywhere

some bugs to fix

fixed
--- some things are sorted as strings
--- try-except to get max from empty table
--- chat names with spaces in them get {bereza vorona}
--- instead of query1 + table_name + query2, something more clever

left to fix
--- instead of looking through chats one after another, PROGRESSBAR, somehow get them all (study VK API)
--- works really slow, why? maybe disable autocommit
### now works better because of less timeout
--- something else

s17b2-voroneckij commented on start_me.py:
```diff
@@ -220,18 +206,17 @@ def show_main_window():
     main.mainloop()
 
 
 ncb = 0
```
Всё это в if __name__ == "__main__"

ilya-pauzner committed:
Create README.md

Commit:
--minor interface changes
--stop logging "to_put"
--less global variables
--same format for SQL requests

TwoBlueCats commented on start_me.py:
```diff
@@ -82,185 +93,139 @@ def build_auth_window():
+        border = 0
+    else:
+        border = int(border)
+    bad = False
```
Сделать имя флага более понятным

TwoBlueCats commented on start_me.py:
```diff
@@ -82,185 +93,139 @@ def build_auth_window():
+    if not list(connection.execute("SELECT name FROM sqlite_master WHERE type='table' AND name=%s" % table_name)):
+        # first-time
+        connection.execute('CREATE TABLE %s (message_id integer, user_id integer, length integer)' % table_name)
+    border = unpack_value(connection.execute('SELECT MAX(message_id) FROM %s' % table_name))
```
Возможно есть более красивый и главное работающий способ извлекать первый элемент

TwoBlueCats commented on start_me.py:
```diff
@@ -82,185 +93,139 @@ def build_auth_window():
-    for i in range(1, 100):
-        update(i)
+    global api, connection
+    cur = unpack_value(connection.execute('SELECT MAX(id) FROM chatid_to_chatname'))
```
Возможно есть более красивый и главное работающий способ извлекать первый элемент

TwoBlueCats commented on start_me.py:
```diff
@@ -82,185 +93,139 @@ def build_auth_window():
+        cur = int(cur) + 1
+    while True:
+        try:
+            temp = api.messages.getChat(chat_id=cur)
```
Более осмысленное название переменной

TwoBlueCats commented on start_me.py:
```diff
@@ -82,185 +93,139 @@ def build_auth_window():
 logging.basicConfig(format=u'%(filename)s[LINE:%(lineno)d]# %(levelname)-8s [%(asctime)s]  %(message)s',
                     level=logging.INFO)
-
+root = None
```
Возможно стоит сделать отдельный singleton класс, в экземпляре которого хранить все наши глобальные переменные

TwoBlueCats commented on start_me.py:
```diff
@@ -82,185 +93,139 @@ def build_auth_window():
                     level=logging.INFO)
-
+root = None
 api_init()
```
Кажется, что это всё лучше объединить в функцию main или засунуть внутрь ифа
